examples/pytorch/rgcn/entity_sample_cugraph_relgraphconv.py

```diff
@@ -0,0 +1,184 @@
+    # construct sampler and dataloader
+    sampler = MultiLayerNeighborSampler(fanouts)
+    train_loader = DataLoader(
+        g,
+        target_idx[train_idx].type(g.idtype),
+        sampler,
+        device=device,
+        batch_size=100,
+        shuffle=True,
+    )
+    # no separate validation subset, use train index instead for validation
+    val_loader = DataLoader(
+        g,
+        target_idx[train_idx].type(g.idtype),
+        sampler,
+        device=device,
+        batch_size=100,
+        shuffle=False,
+    )
+    model.train()
+    for epoch in range(50):
+        total_loss = 0
+        for it, (input_nodes, output_nodes, blocks) in enumerate(train_loader):
+            output_nodes = inv_target[output_nodes.type(torch.int64)]
+            for block in blocks:
+                block.edata["norm"] = dgl.norm_by_dst(block).unsqueeze(1)
+            logits = model(blocks)
+            loss = loss_fcn(logits, labels[output_nodes])
+            optimizer.zero_grad()
+            loss.backward()
+            optimizer.step()
+            total_loss += loss.item()
+        acc = evaluate(model, labels, val_loader, inv_target)
+        print(
+            "Epoch {:05d} | Loss {:.4f} | Val. Accuracy {:.4f} ".format(
+                epoch, total_loss / (it + 1), acc
+            )
+        )
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="RGCN for entity classification with sampling"
+    )
+    parser.add_argument(
+        "--dataset",
+        type=str,
+        default="aifb",
+        help="Dataset name ('aifb', 'mutag', 'bgs', 'am'), default to 'aifb'.",
+    )
+    parser.add_argument(
+        "--idtype",
+        type=str,
+        default="int64",
+        help="Graph idtype ('int64' or 'int32'), default to 'int64'.",
+    )
+    args = parser.parse_args()
+    device = torch.device("cuda")
+    print(f"Training with DGL CuGraphRelGraphConv module with sampling.")
+
+    # load and preprocess dataset
+    if args.dataset == "aifb":
+        data = AIFBDataset()
+    elif args.dataset == "mutag":
+        data = MUTAGDataset()
+    elif args.dataset == "bgs":
+        data = BGSDataset()
+    elif args.dataset == "am":
+        data = AMDataset()
+    else:
+        raise ValueError("Unknown dataset: {}".format(args.dataset))
+    hg = data[0].to(device)
+    hg = hg.int() if args.idtype == "int32" else hg.long()
+    num_rels = len(hg.canonical_etypes)
+    category = data.predict_category
+    labels = hg.nodes[category].data.pop("labels")
+    train_mask = hg.nodes[category].data.pop("train_mask")
+    test_mask = hg.nodes[category].data.pop("test_mask")
+    # find target category and node id
+    category_id = hg.ntypes.index(category)
+    g = dgl.to_homogeneous(hg)
+    node_ids = torch.arange(g.num_nodes()).to(device)
+    target_idx = node_ids[g.ndata[dgl.NTYPE] == category_id]
+    # rename the field as they can be changed by DataLoader
```

Comment:
Can you be more specific?

Reply:
This was copied from entity_sample directly. Comment has been removed.